book/src/rent.md

```diff
@@ -0,0 +1,52 @@
+# Rent
+
+Accounts on Solana may have owner-controlled state (`Account::data`) that's separate 
+from the account's balance (`Account::lamports`).  Since validators on the network need 
+to maintain a working copy of this state in memory, the network charges a time-and-space 
+based fee for this resource consumption, also known as Rent.
+
+## Two-tiered rent regime
+
+Accounts which maintain a minimum balance equivalent to 2 years of rent payments are exempt.  Accounts whose balance falls below this threshold are charged rent at a rate specified in genesis, in lamports per kilobyte-year.  The network charges rent on a per-epoch 
+basis, in credit for the next epoch (but in arrears when necessary), and `Account::rent_epoch` keeps track of the next time 
```

Comment:
Do we expect the cost of rent to vary over time?   If it's stable, could we instead of `rent_epoch` compute `eviction_epoch`?  The epoch when the account's lamports will not cover rent anymore.   This complicates the rent re-computation when lamports are removed/added to the account but simplifies a future space optimization where at the end of each epoch the node could simply delete any accounts with a lesser `eviction_epoch`.   Just a thought.  Overall the latest design seems real nice

Reply:
Nice idea, but I think it might end up coming out the same as the above. 

I'm trying to avoid any scans of the DB at epoch, slot, or other boundaries.  If it happens you haven't paid rent, your account just vanishes as a result of a load() failure.

In such case, we'd have to check eviction_epoch at each load(), so...

Reply:
> your account just vanishes as a result of a load() failure.

Yep, got that part.  I was just thinking about when we need to add the optimization in the future to periodically scan and remove all those delinquent accounts that nobody will ever access again but are still hanging out, waiting for the next load() that never comes to vanish them.    I don't think this affects the design in any way, just potentially a bit of the implementation (later)